### PR TITLE
`Quiz exercises`: Improve apollon diagram import

### PR DESCRIPTION
**Pull request by milljoniaer**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

#### Client
- [x] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added `authorities` to all new routes and checked the course groups for displaying navigation elements (links, buttons).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context
slightly improves the UX of apollon diagram import for quiz exercises:
- make title required
- disable save/generate button when form not valid
- slightly improve UI

### Steps for Testing

Prerequisites:
- 1 Instructor

1. Log in to Artemis
2. Create a new quiz exercise 
3. Add an Apollon Diagram drag-and-drop question
4. verify that everything works

#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Screenshots
image

## Summary by CodeRabbit

- **New Features**
	- Introduced input validation and mandatory title field for creating diagrams to enhance data integrity.
	- Improved interactive elements' responsiveness in diagram details.
  
- **Enhancements**
	- Updated button text translations for clarity.
	- Enhanced styling for buttons and layout adjustments in diagram listings.

- **Bug Fixes**
	- Adjusted diagram creation handling to improve reliability of diagram save operations.

### CODE REVIEW

**Review by laurenzfb — DISMISSED** [review]
code lgtm

**Review by laurenzfb — APPROVED** [review]
reapprove after fixing tests

**Review by rstief — APPROVED** [review]
Code LGTM, also tested on ts2 and works as expected.

**Review by Jan-Thurner — APPROVED** [review]
Works as expected

**Review by pzdr7 — APPROVED** [review]
Code looks good 👍

File: src/main/webapp/app/exercises/quiz/manage/apollon-diagrams/apollon-diagram-detail.component.html [hunk 1]
```diff
@@ -22,15 +22,23 @@ <h4 class="modal-title">{{ 'artemisApp.apollonDiagram.detail.title' | artemisTra
...
     <button class="btn btn-outline-primary me-auto" (click)="confirmExitDetailView(true)">Close</button>
-    <button class="btn btn-outline-primary" (click)="saveDiagram()" [hidden]="!apollonDiagram" [disabled]="isSaved" jhiTranslate="artemisApp.apollonDiagram.detail.save"></button>
-    <button id="generate-quiz-exercise" class="btn btn-primary" (click)="generateExercise()" [hidden]="!apollonDiagram">
-        {{ 'artemisApp.apollonDiagram.detail.createLabel' | artemisTranslate }}
-    </button>
+    <button
+        class="btn btn-outline-primary"
+        (click)="saveDiagram()"
+        [hidden]="!apollonDiagram"
+        [disabled]="isSaved || (titleField && titleField?.invalid)"
```
Comment (rabeatwork) [inline src/main/webapp/app/exercises/quiz/manage/apollon-diagrams/apollon-diagram-detail.component.html:36]: `titleField?.invalid` should be enough

**Pull request merged**